woocommerce/woocommerce
# Incorporate additional product data to generate AI description

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #38338 

This PR contains updates to improve product description generation. It includes categories, tags, and other product attributes in the prompt and refactors the product data utility's `getAttributes` method to return attribute names instead of their keys. 

I've also tried to optimize the prompt to help reduce the frequency of the LLM overusing the added data (categories, tags, etc.). However, there may still be occasional instances that the input data is overused in the generated description.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new product or edit an existing one
2. Add categories, tags, and attributes to the product
3. Write a title for the product and click "Write it for me" in the description box
4. Confirm that the generated description uses the categories, tags, and other attributes that you've added to the product
